subt/main.py

```diff
@@ -485,7 +488,16 @@ def follow_trace(self, trace, timeout, max_target_distance=5.0, safety_limit=Non
                     if safety_limit is not None and safety < safety_limit:
                         print('Danger! Safety limit for follow trace reached!', safety, safety_limit)
                         break
+
+                if is_trace3d:
+                    d = distance3D(self.xyz, [target_x, target_y, target_z])
+                    time_to_target = d/self.max_speed
+                    desired_z_speed = (target_z - self.xyz[2]) / time_to_target
+                    self.bus.publish('desired_z_speed', desired_z_speed)
```

Comment:
You may need to dig deeper and clear interfaces. Here you are setting vertical speed long after you set horizontal speed. I don't think that go_safely() guarantees it would only send one velocity command. flip() certainly happily sends more. I.e. a lot can happen after you set horizontal velocity and before you set the vertical velocity.

Reply:
Hmm, you are right that if you combine `symmetric` with `is_trace3d` then it would not be good. Also `go_safely` would need to be 3D ...

Reply:
I do not see simple way out of this :(. If we change the planner then the trajectory would more strictly defined, like flying sideways. Also change to X500 may completely change the game - no "front", only lidar and up/down RGBD. Any suggestion? Should I drop it from this PR? (it worked for X4 somehow)

Reply:
Ack, it will need a complete redesign. We can do it separately. Right after we delete configuration for R2 ;-)

Reply:
:) OK

Reply:
https://github.com/robotika/osgar/pull/806